### PR TITLE
[PD 2986] refactor and reskin cbo

### PR DESCRIPTION
**Pull request by koonsies**
## What's this PR do?

Too many things.

Just kidding. It refactors a bunch of code to bring it in line with the way we write out React code. 

Huy didn't write bad code, but he wrote it in the way a greenfield project would have been written. None of our reusable components or installed packages were used, and much of the code was hard to read for someone used to working with our existing solutions.

## Where should the reviewer start?

The beta menu should contain an entry for CBO. Click that.

## How should this be manually tested?

Once through the beta menu, test the edit and delete functionality of the forms. 

## Any background context you want to provide?

Most of the data flow of this app revolves around the class "CboZipper". This class allows the developer to store a zipper alongside the relevant data. In Redux, data is stored by CBO.. At the core, there is 

``` 
CBOS: [
    {cbo data,
     contacts: [
          {contact data,
            distributionLocations: [
                  {distribution location data}
           }
           ....
           ]
      }
      ....
      ]  
```

I hope that was nice to read.. But essentially, CboZipper maintains the hierarchical data for a CBO as well as references to a specific contact or distribution location if they were provided. It makes updating redux a breeze after an update.

## Scrippzz

AppAccess up in your grill

```
from myjobs.models import AppAccess
from seo.models import Company

de = Company.objects.get(name='DirectEmployers Association') # Change DirectEmployers to whichever company you want to use.
aa = AppAccess.objects.get(name='cbo')

de.app_access.add(aa)
```

### CODE REVIEW

File: cbo/api.py [hunk 1]
```diff
@@ -0,0 +1,501 @@
+import json
```
Comment (koonsies) [inline cbo/api.py]: Behold: Sample data

**Comment by koonsies** [comment]
Putting up DNM until I get the permissions migration added. Joy.

File: cbo/migrations/0001_initial_permissions_app_access.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.10 on 2017-03-10 16:14
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from myjobs.models import AppAccess, Activity
+
+
+def forwards(apps, schema_editor):
```
Comment (darrint) [inline cbo/migrations/0001_initial_permissions_app_access.py]: You are missing a decorator we use for other RunPython migrations.

**Comment by koonsies** [comment]
I'm missing something. Anyone have an idea why this import is failing?

20:42:27 [1m[31mERROR in ./src/cbo/components/App.js
20:42:27 Module not found: Error: Cannot resolve 'file' or 'directory' ../../common/ui/Loading in /home/ubuntu/workspace/PR-Unit-Tests/MyJobs/gulp/src/cbo/components

**Comment by tdphillips** [comment]
Marcus moved that to LoadingSpinner.

**Comment by koonsies** [comment]
Ah, excellent. Thank you! I thought I had merged quality control before putting up the PR buttt... I guess I didn't.

**Pull request merged**